packages/client/src/index.ts

```diff
@@ -203,15 +204,17 @@ export class Revision {
  * ⚠️ Name records are not _yet_ published to or updated from the IPFS network.
  * Working with name records simply updates the w3name cache of data.
  */
-export async function publish (service: W3NameService, revision: Revision, key: PrivateKey) {
+export async function publish (revision: Revision, key: PrivateKey, service: W3NameService = defaultService) {
   const url = new URL(`name/${revision.name.toString()}`, service.endpoint)
   const entry = await ipns.create(
     key,
     uint8ArrayFromString(revision.value),
     revision.sequence,
     new Date(revision.validity).getTime() - Date.now()
   )
-
+  if (service.rateLimiter) {
+    await service.rateLimiter()
+  }
```

Comment:
I feel that it would be slightly nicer to have this `if` logic handled inside the service class. So that you can just call `await service.waitForRateLimit()` and it deals with any necessary logic based on whether or not it has a ratelimiter.

Reply:
This is now done.